tutorials/plot_background_statistics.py

```diff
@@ -0,0 +1,566 @@
+# test -- namely controlling FWER and making no assumptions of parametric
+# data distribution.
+# Cricitally, though, it also accounts for the correlation structure in the
+# data -- which in this toy case is spatial but
+# in general can be multidimensional (e.g., spatio-temporal) -- because the
+# null distribution will be derived from data in a way that preserves these
+# correlations.
+#
+# However, there is a drawback. If a cluster significantly deviates from
+# the null, no further inference on the cluster (e.g., peak location) can be
+# made, as the entire cluster as a whole is used to reject the null.
+
+# We need to define our connectivity/neighbor/adjacency matrix, which we know
+# is just a grid. Using ``connectivity=None`` is highly optimized for this
+# case.
+
+# Reshape to what is equivalent to (n_samples, n_space, n_time)
+titles.append('Clustering')
+X.shape = (n_subjects, width, width)
+t_clust, clusters, p_values, H0 = permutation_cluster_1samp_test(
+    X, n_jobs=1, threshold=threshold, connectivity=None, tail=1,
+    n_permutations=n_permutations)
+# Put the cluster data in a viewable format
+p_clust = np.ones((width, width))
+for cl, p in zip(clusters, p_values):
+    p_clust[cl] = p
+ts.append(t_clust)
+ps.append(p_clust)
+mccs.append(True)
+plot_t_p(ts[-1], ps[-1], titles[-1], mccs[-1])
+
+###############################################################################
+# "hat" variance adjustment
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+# This method can also be used in this context to correct for small
+# variances [1]_:
+titles.append(r'$\mathbf{C_{hat}}$')
+stat_fun_hat = partial(ttest_1samp_no_p, sigma=sigma)
+t_hat, clusters, p_values, H0 = permutation_cluster_1samp_test(
+    X, n_jobs=1, threshold=threshold, connectivity=None, tail=1,
+    n_permutations=n_permutations, stat_fun=stat_fun_hat, buffer_size=None)
+p_hat = np.ones((width, width))
+for cl, p in zip(clusters, p_values):
+    p_hat[cl] = p
+ts.append(t_hat)
+ps.append(p_hat)
+mccs.append(True)
+plot_t_p(ts[-1], ps[-1], titles[-1], mccs[-1])
+
+###############################################################################
+# .. _tfce_example:
+#
+# Threshold-free cluster enhancement (TFCE)
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# TFCE eliminates the free parameter initial ``threshold`` value that
+# determines which points are included in clustering by approximating
+# a continuous integration across possible threshold values with a standard
+# `Riemann sum <https://en.wikipedia.org/wiki/Riemann_sum>`_ [2]_.
+# This requires giving a starting threshold ``'start'`` and a step
+# size ``'step'``, which in MNE is supplied as a dict.
+# The smaller the ``'step'`` and closer to 0 the ``'start'`` value,
+# the better the approximation, but the longer it takes).
+#
+# A significant advantage of TFCE is that, rather than modifying the
+# statistical test, it modifies the data itself (while still controlling
```

Comment:
To be honest, I'm not sure I understand it, phrased like this.

Although, as mentioned, math impaired.